pyvrp/cpp/TimeWindowSegment.h

```diff
@@ -41,13 +41,14 @@ class TimeWindowSegment
 {
     using TWS = TimeWindowSegment;
 
-    size_t idxFirst_ = 0;       // Index of the first client in the segment
-    size_t idxLast_ = 0;        // Index of the last client in the segment
-    Duration duration_ = 0;     // Total duration, incl. waiting and servicing
-    Duration timeWarp_ = 0;     // Cumulative time warp
-    Duration twEarly_ = 0;      // Earliest visit moment of first client
-    Duration twLate_ = 0;       // Latest visit moment of first client
-    Duration releaseTime_ = 0;  // Earliest allowed moment to leave the depot
+    size_t idxFirst_;       // Index of the first client in the segment
+    size_t idxLast_;        // Index of the last client in the segment
+    Duration duration_;     // Total duration, incl. waiting and servicing
+    Duration timeWarp_;     // Cumulative time warp
+    Duration twEarly_;      // Earliest visit moment of first client
+    Duration twLate_;       // Latest visit moment of first client
+    Duration releaseTime_;  // Earliest allowed moment to leave the depot
+    Duration maxDuration_;  // Maximum segment duration
```

Comment:
> or even use `TWS` to compute duration but handle the maximum duration outside `TWS`?

I'll try this. The alternative of passing an argument to `totalTimeWarp()` makes that property more complex, and I would like to avoid doing that.

Reply:
> The alternative of passing an argument to `totalTimeWarp()` makes that property more complex, and I would like to avoid doing that.

I didn't see a clean way to do so without blowing up everything else, so this is where we're at now. Might refactor later if we can make the evaluation logic less bloated first.